tiledb/sm/rest/curl.cc

```diff
@@ -565,6 +602,7 @@ Status Curl::get_data(
       curl_slist_free_all(headers));
 
   /* pass our list of custom made headers */
+  curl_easy_setopt(curl, CURLOPT_HEADER, 1);
```

Comment:
This is redundant with L220, please remove it. We can set this once because the value will never change, unlike `headers` on L606.